### ISSUE TITLE
Root path with spaces in it are not recognized

### ISSUE THREAD

**Issue by kftb** [Salvoxia/immich-folder-album-creator#30]
Hi! 

Trying to get this amazing script going but I have issues with the root path. Since it is an external library, the folder names unfortunately have spaces in them. I haven't been able to get it to work correctly. 

Tried: 
Putting the path in quotes: 
` ROOT_PATH: "/pictures/00_Pictures_MASTER/00_Sortierte Bilder/"` - It completely ignores the root path and creates folder name with the whole name in it (example: "pictures 00_Pictures_MASTER 00_Sortierte 2009" (note the missing "Bilder" after "Sortierte" where the space is)

Without the quotes:
` ROOT_PATH: /pictures/00_Pictures_MASTER/00_Sortierte Bilder/` - It does the same and ignores the root path (example album: " pictures 00_Pictures_MASTER 00_Sortierte 2009")

Without the folder with a space in it
` ROOT_PATH: /pictures/00_Pictures_MASTER/` - this one works as expected and creates paths from this folder onwards (example album: "00_Sortierte Bilder 2005")

If I try to just put the folder with the spaces in quotes: 
      `ROOT_PATH: /pictures/00_Pictures_MASTER/"00_Sortierte Bilder"/ ` the script fails completely. 

same if I try to escape it: 
      `ROOT_PATH: /pictures/00_Pictures_MASTER/00_Sortierte\ Bilder/ ` 

Am I missing something obvious? I would rather not have to rename folders to avoid spaces.

**Comment by Salvoxia** [comment]
Hi,

I did find an issue with parsing of the `ROOT_PATH` environment variable when using the Docker image.
I just pushed a fix for that to the `main` branch.
Could you please test if using the `latest` image tag resolves your issue? Remember to force pull if you have used the `latest` tag before.

Best Regards,
Salvoxia

**Comment by kftb** [comment]
Hi! Thank you for taking a look so quickly and for pushing the fix. It's working now perfectly.

**Comment by kftb** [comment]
Hi! Thank you for taking a look so quickly and for pushing the fix. It's working now perfectly.

**Comment by Arnoud-B** [comment]
I'm running into the same issue with the Bare Python Script.

**Comment by Salvoxia** [comment]
Can you elaborate on how exactly your setup looks and what your call to the script looks like?
I just tried the following:
I mounted my images to `/external_libs/Immich Test` to the Immich container to create a root path with a blank, added that path to an external lib and scanned the library.

Then I ran
```shell
python3 ./immich_auto_album.py -t file "/external_libs/Immich Test" https://myimmichserver/api api_key.secret
```
It found the images and proposed the correct album names matching the folders contained in `/external_libs/Immich Test`.
So it worked as expected.

**Comment by Arnoud-B** [comment]
When I run it with a space in the folder name, it returns 0 albums, but when I remove the space temporarily, it finds 2 albums. Prompt:
`(venv) sh-4.4$ python3.9 immich_auto_album.py -r "/mnt/media/To Process/2022" -u -a 1 -s - -c 500 -C 500 -k true -m CREATE [http://localhost:2284/api](http://localhost:2284/api) API_KEY`
Result:
time=2024-12-01T21:58:17.545+01:00 level=INFO msg=Detected Immich server version 1.121.0
time=2024-12-01T21:58:17.545+01:00 level=INFO msg=Requesting all assets
time=2024-12-01T21:58:20.042+01:00 level=INFO msg=3682 photos found
time=2024-12-01T21:58:20.042+01:00 level=INFO msg=Sorting assets to corresponding albums using folder name
time=2024-12-01T21:58:20.061+01:00 level=INFO msg=0 albums identified
time=2024-12-01T21:58:20.061+01:00 level=INFO msg=Album list: []
time=2024-12-01T21:58:20.061+01:00 level=INFO msg=Listing existing albums on immich
time=2024-12-01T21:58:22.021+01:00 level=INFO msg=272 existing albums identified
time=2024-12-01T21:58:22.021+01:00 level=INFO msg=Creating albums if needed
time=2024-12-01T21:58:22.022+01:00 level=INFO msg=0 albums created
time=2024-12-01T21:58:22.022+01:00 level=INFO msg=Adding assets to albums
time=2024-12-01T21:58:22.022+01:00 level=INFO msg=Done!

**Comment by Salvoxia** [comment]
I'm a little bit puzzled since you seem to be calling the script with just 2 positional arguments, whereas it requires 3.
Can you run that with `-l DEBUG` and share the first part with the script logging all the passed arguments?
```
time=2024-12-01T22:08:07.022+01:00 level=DEBUG msg=root_path = ['/external_libs/Immich Test']
time=2024-12-01T22:08:07.023+01:00 level=DEBUG msg=root_url = https://myimmichserver/api
time=2024-12-01T22:08:07.023+01:00 level=DEBUG msg=api_key = myapikey
time=2024-12-01T22:08:07.023+01:00 level=DEBUG msg=number_of_images_per_request = 2000
time=2024-12-01T22:08:07.023+01:00 level=DEBUG msg=number_of_assets_to_fetch_per_request = 5000
time=2024-12-01T22:08:07.023+01:00 level=DEBUG msg=unattended = False
time=2024-12-01T22:08:07.023+01:00 level=DEBUG msg=album_levels = 1
time=2024-12-01T22:08:07.023+01:00 level=DEBUG msg=album_level_separator =  
time=2024-12-01T22:08:07.023+01:00 level=DEBUG msg=album_order = False
time=2024-12-01T22:08:07.023+01:00 level=DEBUG msg=insecure = False
time=2024-12-01T22:08:07.023+01:00 level=DEBUG msg=ignore = None
time=2024-12-01T22:08:07.024+01:00 level=DEBUG msg=mode = CREATE
time=2024-12-01T22:08:07.024+01:00 level=DEBUG msg=delete_confirm = False
time=2024-12-01T22:08:07.024+01:00 level=DEBUG msg=is_docker = False
time=2024-12-01T22:08:07.024+01:00 level=DEBUG msg=share_with = None
time=2024-12-01T22:08:07.024+01:00 level=DEBUG msg=share_role = viewer
time=2024-12-01T22:08:07.024+01:00 level=DEBUG msg=sync_mode = 0
time=2024-12-01T22:08:07.024+01:00 level=DEBUG msg=find_assets_in_albums = False
time=2024-12-01T22:08:07.024+01:00 level=DEBUG msg=path_filter = None
time=2024-12-01T22:08:07.024+01:00 level=DEBUG msg=set_album_thumbnail = None
time=2024-12-01T22:08:07.024+01:00 level=DEBUG msg=archive = False
time=2024-12-01T22:08:07.024+01:00 level=DEBUG msg=find_archived_assets = False
```

**Comment by Arnoud-B** [comment]
With DEBUG:
time=2024-12-01T22:12:33.036+01:00 level=DEBUG msg=root_path = ['/mnt/media//To Process/2022', 'true']
time=2024-12-01T22:12:33.036+01:00 level=DEBUG msg=root_url = http://localhost:2284/api
time=2024-12-01T22:12:33.036+01:00 level=DEBUG msg=api_key = API_KEY
time=2024-12-01T22:12:33.037+01:00 level=DEBUG msg=number_of_images_per_request = 500
time=2024-12-01T22:12:33.037+01:00 level=DEBUG msg=number_of_assets_to_fetch_per_request = 500
time=2024-12-01T22:12:33.037+01:00 level=DEBUG msg=unattended = True
time=2024-12-01T22:12:33.037+01:00 level=DEBUG msg=album_levels = 1
time=2024-12-01T22:12:33.037+01:00 level=DEBUG msg=album_level_separator = -
time=2024-12-01T22:12:33.037+01:00 level=DEBUG msg=album_order = False
time=2024-12-01T22:12:33.037+01:00 level=DEBUG msg=insecure = True
time=2024-12-01T22:12:33.038+01:00 level=DEBUG msg=ignore = None
time=2024-12-01T22:12:33.038+01:00 level=DEBUG msg=mode = CREATE
time=2024-12-01T22:12:33.038+01:00 level=DEBUG msg=delete_confirm = False
time=2024-12-01T22:12:33.038+01:00 level=DEBUG msg=is_docker = False
time=2024-12-01T22:12:33.038+01:00 level=DEBUG msg=share_with = None
time=2024-12-01T22:12:33.038+01:00 level=DEBUG msg=share_role = viewer
time=2024-12-01T22:12:33.038+01:00 level=DEBUG msg=sync_mode = 0
time=2024-12-01T22:12:33.038+01:00 level=DEBUG msg=find_assets_in_albums = False
time=2024-12-01T22:12:33.039+01:00 level=DEBUG msg=path_filter = None
time=2024-12-01T22:12:33.039+01:00 level=DEBUG msg=set_album_thumbnail = None
time=2024-12-01T22:12:33.039+01:00 level=DEBUG msg=archive = False
time=2024-12-01T22:12:33.039+01:00 level=DEBUG msg=find_archived_assets = False
time=2024-12-01T22:12:33.043+01:00 level=DEBUG msg=Starting new HTTP connection (1): localhost:2284
time=2024-12-01T22:12:33.055+01:00 level=DEBUG msg=http://localhost:2284 "GET /api/server/version HTTP/11" 200 33
time=2024-12-01T22:12:33.056+01:00 level=INFO msg=Detected Immich server version 1.121.0
time=2024-12-01T22:12:33.057+01:00 level=INFO msg=Requesting all assets
time=2024-12-01T22:12:33.059+01:00 level=DEBUG msg=Starting new HTTP connection (1): localhost:2284
time=2024-12-01T22:12:33.079+01:00 level=DEBUG msg=http://localhost:2284 "POST /api/search/metadata HTTP/11" 401 96

**Comment by Salvoxia** [comment]
`time=2024-12-01T22:12:33.036+01:00 level=DEBUG msg=root_path = ['/mnt/media//To Process/2022', 'true']`
Looks strange. Double slashes, and a `true` interpreted as a path.
Can you please try calling the script like this?
`python3.9 immich_auto_album.py -u -a 1 -s " - " -c 500 -C 500 -k "/mnt/media/To Process/2022" http://localhost:2284/api API_KEY`

The documentation for `-k / --insecure` seems to be wrong, which might have contributed to this situation. To disable SSL verification, the option should be passed, but without a value.
`-k, --insecure        Set to true to ignore SSL verification (default: False)`
should be
`-k, --insecure       Pass to ignore SSL verification`

**Comment by Arnoud-B** [comment]
Result with your suggestion & `-m CREATE`:
```
time=2024-12-01T22:25:53.306+01:00 level=INFO msg=Detected Immich server version 1.121.0
time=2024-12-01T22:25:53.307+01:00 level=INFO msg=Requesting all assets
time=2024-12-01T22:25:55.483+01:00 level=INFO msg=3682 photos found
time=2024-12-01T22:25:55.483+01:00 level=INFO msg=Sorting assets to corresponding albums using folder name
time=2024-12-01T22:25:55.498+01:00 level=INFO msg=0 albums identified
time=2024-12-01T22:25:55.499+01:00 level=INFO msg=Album list: []
time=2024-12-01T22:25:55.499+01:00 level=INFO msg=Listing existing albums on immich
time=2024-12-01T22:25:57.644+01:00 level=INFO msg=272 existing albums identified
time=2024-12-01T22:25:57.644+01:00 level=INFO msg=Creating albums if needed
time=2024-12-01T22:25:57.644+01:00 level=INFO msg=0 albums created
time=2024-12-01T22:25:57.645+01:00 level=INFO msg=Adding assets to albums
time=2024-12-01T22:25:57.645+01:00 level=INFO msg=Done!
```
Result when I remove the space, same command:
```
time=2024-12-01T22:27:05.415+01:00 level=INFO msg=Detected Immich server version 1.121.0
time=2024-12-01T22:27:05.415+01:00 level=INFO msg=Requesting all assets
time=2024-12-01T22:27:10.017+01:00 level=INFO msg=3682 photos found
time=2024-12-01T22:27:10.017+01:00 level=INFO msg=Sorting assets to corresponding albums using folder name
time=2024-12-01T22:27:10.041+01:00 level=INFO msg=2 albums identified
time=2024-12-01T22:27:10.041+01:00 level=INFO msg=Album list: ['FOLDER1', 'FOLDER2']
time=2024-12-01T22:27:10.041+01:00 level=INFO msg=Listing existing albums on immich
time=2024-12-01T22:27:16.139+01:00 level=INFO msg=272 existing albums identified
time=2024-12-01T22:27:16.140+01:00 level=INFO msg=Creating albums if needed
time=2024-12-01T22:27:16.140+01:00 level=INFO msg=0 albums created
time=2024-12-01T22:27:16.140+01:00 level=INFO msg=Adding assets to albums
time=2024-12-01T22:27:16.596+01:00 level=INFO msg=285 new assets added to FOLDER1
time=2024-12-01T22:27:16.945+01:00 level=INFO msg=274 new assets added to FOLDER2
time=2024-12-01T22:27:16.946+01:00 level=INFO msg=Done!
```

**Comment by Salvoxia** [comment]
Could you please run my command again with `-l DEBUG`? :)

**Comment by Arnoud-B** [comment]
Result with your exact command with `-l DEBUG`:
```
time=2024-12-01T22:33:20.079+01:00 level=DEBUG msg=root_path = ['/mnt/media/To Process/2022']
time=2024-12-01T22:33:20.079+01:00 level=DEBUG msg=root_url = http://localhost:2284/api
time=2024-12-01T22:33:20.079+01:00 level=DEBUG msg=api_key = API_KEY
time=2024-12-01T22:33:20.079+01:00 level=DEBUG msg=number_of_images_per_request = 500
time=2024-12-01T22:33:20.079+01:00 level=DEBUG msg=number_of_assets_to_fetch_per_request = 500
time=2024-12-01T22:33:20.080+01:00 level=DEBUG msg=unattended = True
time=2024-12-01T22:33:20.080+01:00 level=DEBUG msg=album_levels = 1
time=2024-12-01T22:33:20.080+01:00 level=DEBUG msg=album_level_separator =  - 
time=2024-12-01T22:33:20.080+01:00 level=DEBUG msg=album_order = False
time=2024-12-01T22:33:20.080+01:00 level=DEBUG msg=insecure = True
time=2024-12-01T22:33:20.080+01:00 level=DEBUG msg=ignore = None
time=2024-12-01T22:33:20.080+01:00 level=DEBUG msg=mode = CREATE
time=2024-12-01T22:33:20.080+01:00 level=DEBUG msg=delete_confirm = False
time=2024-12-01T22:33:20.080+01:00 level=DEBUG msg=is_docker = False
time=2024-12-01T22:33:20.081+01:00 level=DEBUG msg=share_with = None
time=2024-12-01T22:33:20.081+01:00 level=DEBUG msg=share_role = viewer
time=2024-12-01T22:33:20.081+01:00 level=DEBUG msg=sync_mode = 0
time=2024-12-01T22:33:20.081+01:00 level=DEBUG msg=find_assets_in_albums = False
time=2024-12-01T22:33:20.081+01:00 level=DEBUG msg=path_filter = None
time=2024-12-01T22:33:20.081+01:00 level=DEBUG msg=set_album_thumbnail = None
time=2024-12-01T22:33:20.081+01:00 level=DEBUG msg=archive = False
time=2024-12-01T22:33:20.081+01:00 level=DEBUG msg=find_archived_assets = False
time=2024-12-01T22:33:20.085+01:00 level=DEBUG msg=Starting new HTTP connection (1): localhost:2284
time=2024-12-01T22:33:20.113+01:00 level=DEBUG msg=http://localhost:2284 "GET /api/server/version HTTP/11" 200 33
time=2024-12-01T22:33:20.114+01:00 level=INFO msg=Detected Immich server version 1.121.0
time=2024-12-01T22:33:20.115+01:00 level=INFO msg=Requesting all assets
time=2024-12-01T22:33:26.560+01:00 level=INFO msg=3123 photos found
time=2024-12-01T22:33:26.560+01:00 level=INFO msg=Sorting assets to corresponding albums using folder name
time=2024-12-01T22:33:26.877+01:00 level=INFO msg=0 albums identified
time=2024-12-01T22:33:26.877+01:00 level=INFO msg=Album list: []
time=2024-12-01T22:33:26.877+01:00 level=INFO msg=Listing existing albums on immich
time=2024-12-01T22:33:26.881+01:00 level=DEBUG msg=Starting new HTTP connection (1): localhost:2284
time=2024-12-01T22:33:39.231+01:00 level=DEBUG msg=http://localhost:2284 "GET /api/albums HTTP/11" 200 203381
time=2024-12-01T22:33:39.242+01:00 level=INFO msg=272 existing albums identified
time=2024-12-01T22:33:39.242+01:00 level=INFO msg=Creating albums if needed
time=2024-12-01T22:33:39.242+01:00 level=INFO msg=0 albums created
time=2024-12-01T22:33:39.242+01:00 level=INFO msg=Adding assets to albums
time=2024-12-01T22:33:39.242+01:00 level=INFO msg=Done!
```
I removed the chunks and identified root_paths, but what's interesting is that in those root_paths it finds `mnt/media/Process/*` (the folder name without space that I temporarily used) but not `mnt/media/To Process/*`.

**Comment by Salvoxia** [comment]
You need to be careful when switching around the mounting paths for your external library. Immich will not simply pick up the change. [This comment](https://github.com/Salvoxia/immich-folder-album-creator/issues/78#issuecomment-2510144318) basically applies to your situation when switching around the mount paths. 

The latest run looks good, but it seems Immich is not aware of the currently correct `root_path`. Can you verify that please?

**Comment by Arnoud-B** [comment]
The path for the external library in Immich is `/mnt/media/Process`. This is mapped in the container as `- /volume1/homes/USER/Photos/To Process:/mnt/media/Process:ro`.

To use your script, I renamed `To Process` to `Process`, then ran the script to create the albums and add the photos, then renamed the folder back to `To Process`. After that I let Immich do a full rescan and everything seemed to work fine.

To avoid potential problems I'd like to get it to work without renaming though, especially since it's not a one-time usage 😅

**Comment by Salvoxia** [comment]
Ok, I just simulated exactly that:
Mounted my external library path to `/external_libs/Immich Test` in the container, added it as import path to an external library, scanned, let the script run and create albums.
Then I remounted to `/external_libs/ImmichTest`. Everything looks fine until you try to zoom in to one of the images, then I get an image load error since the image no longer exists at the path it originally was discovered at. Immich did for some reason not recognize the asset as "offline".
If you don't zoom into an image, Immich will use thumbnails it generated at import-time, so you might not notice the image is no longer available in the container.

**Comment by Arnoud-B** [comment]
..and I just realized I had the mount path confused with the folder name it was mapped to. Even then, it only worked when I renamed the folder (I checked zooming in the albums that were created earlier this thread (FOLDER1) and there was no issue).

However, I just tested it again with your command and without renaming the folder, and now it works. No idea what changed unfortunately!

Thank you for all your help.
